WebApplication1/WebApplication1/main.js

```diff
@@ -0,0 +1,81 @@
+﻿
+
+var csv = {
+    read: function (location) {
+
+        var user_lat = location.coords.latitude;
+        var user_lon = location.coords.longitude;
+
+        var rawFile = new XMLHttpRequest();
+        rawFile.open("GET", "coffee_shops.csv", true);
+        rawFile.onreadystatechange = function () {
+            if (rawFile.readyState === 4) {
+                if (rawFile.status === 200 || rawFile.status == 0) {
```

Comment:
The if's can be collapsed into one if, and it can be done even nicer if you extract a `readSuccess` method.

All the code from this method is in the body of the if, please come up with a better way, there is no need to got into the function if the file was not read.